A32NX/html_ui/Pages/A32NX_Core/A32NX_ADIRS.js

```diff
@@ -1,65 +1,126 @@
-            SimVar.SetSimVarValue("L:A32NX_ADIRS_PFD_ALIGNED_ATT", "Bool", 0);
-            ADIRSState = 0;
-        }
+        if (this.knob === 0) {
+            this.timer = -1;
+            SimVar.SetSimVarValue(`L:A32NX_ADIRS_ALIGN_LIGHT_${this.num}`, 'Bool', 0);
+        } else {
+            if (this.timer === -1) {
+                // Align time changes depending on latitude.
+                const currentLatitude = SimVar.GetSimVarValue("GPS POSITION LAT", "degree latitude");
+                this.timer = ((currentLatitude ** 2) * 0.095) + 310;
 
-        if (AllADIRSOn && ADIRSState == 0) {
-            //Start ADIRS Alignment
-            SimVar.SetSimVarValue("L:A320_Neo_ADIRS_STATE", "Enum", 1);
-            SimVar.SetSimVarValue("L:A320_Neo_ADIRS_IN_ALIGN", "Bool", 1); // DEFAULT ALIGN ON GPS
-            SimVar.SetSimVarValue("L:A32NX_ADIRS_PFD_ALIGNED_FIRST", "Bool", 0);
-            SimVar.SetSimVarValue("L:A32NX_ADIRS_PFD_ALIGNED_ATT", "Bool", 0);
-            ADIRSState = 1;
-            const currentLatitude = SimVar.GetSimVarValue("GPS POSITION LAT", "degree latitude");
-            ADIRSTimer = (Math.pow(currentLatitude, 2) * 0.095) + 310; // ADIRS ALIGN TIME DEPENDING ON LATITUDE.
-            SimVar.SetSimVarValue("L:A320_Neo_ADIRS_TIME", "Seconds", ADIRSTimer);
-            SimVar.SetSimVarValue("L:A32NX_Neo_ADIRS_START_TIME", "Seconds", ADIRSTimer);
-        }
+                SimVar.SetSimVarValue(`L:A32NX_ADIRS_ALIGN_LIGHT_${this.num}`, 'Bool', 1);
 
-        if (ADIRSState == 1 && SimVar.GetSimVarValue("L:A320_Neo_ADIRS_IN_ALIGN", "Bool") == 1) {
-            if (ADIRSTimer > 0) {
-                ADIRSTimer -= deltaTime / 1000;
-                SimVar.SetSimVarValue("L:A320_Neo_ADIRS_TIME", "Seconds", ADIRSTimer);
-                const ADIRSTimerStartTime = SimVar.GetSimVarValue("L:A32NX_Neo_ADIRS_START_TIME", "Seconds");
-                const secondsIntoAlignment = ADIRSTimerStartTime - ADIRSTimer;
-                if (SimVar.GetSimVarValue("L:A32NX_ADIRS_PFD_ALIGNED_FIRST", "Bool") == 0 && secondsIntoAlignment > 18) {
-                    SimVar.SetSimVarValue("L:A32NX_ADIRS_PFD_ALIGNED_FIRST", "Bool", 1);
-                }
-                if (SimVar.GetSimVarValue("L:A32NX_ADIRS_PFD_ALIGNED_ATT", "Bool") == 0 && secondsIntoAlignment > 28) {
-                    SimVar.SetSimVarValue("L:A32NX_ADIRS_PFD_ALIGNED_ATT", "Bool", 1);
-                }
-                if (ADIRSTimer <= 0) {
-                    //ADIRS Alignment Completed
-                    SimVar.SetSimVarValue("L:A320_Neo_ADIRS_STATE", "Enum", 2);
-                    SimVar.SetSimVarValue("L:A320_Neo_ADIRS_IN_ALIGN", "Bool", 0);
+                // Flash FAULT light
+                SimVar.SetSimVarValue(`L:A32NX_ADIRS_${this.num}_FAULT`, 'Bool', 1);
+                setTimeout(() => {
+                    SimVar.SetSimVarValue(`L:A32NX_ADIRS_${this.num}_FAULT`, 'Bool', 0);
+                }, 100);
```

Comment:
Cool! The flash in the reference is a little longer actually - stepping through frame by frame it's actually two flashes, at around 4-6 frames each and the entire sequence with power being 13-16 frames (plus a few frames afterglow). At 30 FPS, that's around half a second.

So we could match the reference a bit closer here by either increasing the timeout to 500ms, or alternatively going all the way and introducing the double flash with something like 2*200ms and ~100ms break between. As I don't think there's any afterglow in the sim these numbers might need a bit of tweaking to look accurate though, maybe a little shorter gives a better effect.

There's also a very short delay after the switch is turned and the flash starting, but that's probably negligible.

Reply:
Can you link me the reference here or in discord?

Reply:
Ah, I thought this was in the opening comment, sorry. I just looked closely at this video from the second comment: https://youtu.be/o3--ALDKLUc at around 30s. There is some variation between the units, but it's pretty minimal. If we can find more video footage we can cross check further.

Reply:
Look at this video, it has a the same ADIRS that is in our model. The light sequence seem to be pretty different from the older one. https://www.youtube.com/watch?v=Mo7A3CuVb8g

Reply:
Interesting, in that video only the IR fault flight flashes, not the ADR fault light. Not sure which one this commit adds from looking at the code alone. Anyway, for timing reference there it only appears for a single frame instantly with the knob being turned. As that's a 60FPS video that means the light is only on for 16-17ms or so.

Reply:
This one adds the IR fault light. Sadly in my video, the ADR button is covered by his hand so no idea if it flashes or not.

Reply:
So if it's the same light then and the video is from the same type of unit, then the current implementation is mostly good, except that the timeout could probably be reduced to the smallest possible time that allows a flash to be well visible at a reasonable frame rate. Maybe 100ms isn't so far off then after all, but a bit less might still work.